tracy-client-sys/build.rs

```diff
@@ -48,25 +48,32 @@ fn set_feature_defines(mut c: cc::Build) -> cc::Build {
 }
 
 fn main() {
-    if std::env::var_os("CARGO_FEATURE_ENABLE").is_some() {
-        set_feature_defines(cc::Build::new())
-            .file("tracy/TracyClient.cpp")
-            .define("TRACY_MANUAL_LIFETIME", None)
-            .define("TRACY_DELAYED_INIT", None)
-            .warnings(false)
-            .cpp(true)
-            .flag_if_supported("-std=c++11")
-            .compile("libtracy-client.a");
-    }
-
-    match std::env::var("CARGO_CFG_TARGET_OS") {
-        Ok(target_os) => {
-            link_libraries(&target_os);
+    if let Ok(lib) = std::env::var("TRACY_CLIENT_LIB") {
+        if let Ok(lib_path) = std::env::var("TRACY_CLIENT_LIB_PATH") {
+            println!("cargo:rustc-link-search={}", lib_path);
         }
-        Err(e) => {
-            writeln!(::std::io::stderr(), "Unable to get target_os=`{}`!", e)
-                .expect("could not report the error");
-            ::std::process::exit(0xfd);
+        println!("cargo:rustc-link-lib={}", lib);
```

Comment:
Needs an ability to specify static vs dynamic linking, and should explicitly specify the mode here. I don’t if it makes a ton of sense to statically bundle-link the client libraries, so linking to dynamic libraries should probably be a default.

Reply:
> Needs an ability to specify static vs dynamic linking

Normally this problem is solved by providing only the desired form of library in the search path.

Reply:
Unfortunately traditional unixes typically just chuck all types and kinds of libraries into the same directory (`/usr/lib` or some triple-specific directory underneath).

Reply:
If you want a different form than your OS packages, you can place it elsewhere and supply the path explicitly, which should give it precedence. I'm not sure I've ever seen an environment where it's idiomatic for packages to provide both static and dynamic versions of everything, and even then you could still use custom builds.

Reply:
Makes sense for non-windows platforms since they have different extensions (.so/.a).
I can add a `TRACY_CLIENT_STATIC` to switch from dynamic (default) to static, again following openssl-sys a bit since they have the same

Reply:
I’d be happy with `TRACY_CLIENT_STATIC` or whatever if it means rustc is told explicitly which one to link.